Publish builds on `main` branch are failing with a config cache error.

The publish snapshot job on `main` is failing with a config cache error. 

(looks like it is still publishing correctly) 
https://oss.sonatype.org/content/repositories/snapshots/com/slack/lint/compose/compose-lint-checks/1.4.0-SNAPSHOT/

Maybe just use `--no-configuration-cache` for the snapshot publishing? 

It makes it harder to tell at a glance if there is a "real" failure or not. 

https://github.com/slackhq/compose-lints/actions/runs/10647621460/job/29515766042#step:7:35
```
FAILURE: Build failed with an exception.

* What went wrong:
Configuration cache problems found in this build.

3 problems were found storing the configuration cache, 2 of which seem unique.
- Task `:compose-lint-checks:dokkaHtml` of type `org.jetbrains.dokka.gradle.DokkaTask`: cannot serialize object of type 'org.gradle.api.internal.artifacts.configurations.DefaultUnlockedConfiguration', a subtype of 'org.gradle.api.artifacts.Configuration', as these are not supported with the configuration cache.
  See https://docs.gradle.org/8.10/userguide/configuration_cache.html#config_cache:requirements:disallowed_types
- Task `:compose-lint-checks:mavenDokkaJavadocJar` of type `com.vanniktech.maven.publish.tasks.JavadocJar`: cannot serialize object of type 'org.jetbrains.dokka.gradle.DokkaTask', a subtype of 'org.gradle.api.Task', as these are not supported with the configuration cache.
  See https://docs.gradle.org/8.10/userguide/configuration_cache.html#config_cache:requirements:task_access

See the complete report at file:///home/runner/work/compose-lints/compose-lints/build/reports/configuration-cache/6ypudivk1lunb9zflx95ggxnr/9ohkz4ychv2thj2q4bdej83ie/configuration-cache-report.html

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org./

BUILD FAILED in 5s
```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -22,7 +22,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        useK2Uast: [true, false]
+        useK2Uast: [ true, false ]
     steps:
       - name: Checkout
         uses: actions/checkout@v4
@@ -47,6 +47,23 @@ jobs:
           path: |
             **/build/reports/*
 
-      - name: Publish snapshot (main branch only)
-        if: github.repository == 'slackhq/compose-lints' && github.ref == 'refs/heads/main'
-        run: ./gradlew publish -PmavenCentralUsername=${{ secrets.SONATYPEUSERNAME }} -PmavenCentralPassword=${{ secrets.SONATYPEPASSWORD }}
+  publish-snapshots:
+    name: "Publish snapshots"
+    if: github.repository == 'slackhq/compose-lints' && github.ref == 'refs/heads/main'
+    needs: 'build'
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Install JDK
+        uses: actions/setup-java@v4
+        with:
+          distribution: 'zulu'
+          java-version: '21'
+
+      - name: Setup Gradle
+        uses: gradle/actions/setup-gradle@v4
+
+      - name: Publish
+        run: ./gradlew publish -PmavenCentralUsername=${{ secrets.SONATYPEUSERNAME }} -PmavenCentralPassword=${{ secrets.SONATYPEPASSWORD }} --no-configuration-cache

